presentation/src/main/java/com/thanht/foodyentrytask/util/ListDiffAdapter.kt

```diff
@@ -0,0 +1,64 @@
+package com.thanht.foodyentrytask.util
+
+import android.util.Log
+import androidx.recyclerview.widget.DiffUtil
+import androidx.recyclerview.widget.RecyclerView
+import kotlinx.coroutines.*
+
+abstract class ListDiffAdapter<T, H : RecyclerView.ViewHolder> : RecyclerView.Adapter<H>() {
```

Comment:
haha don't use that, try to use `ListAdapter` from `Google` it's simple to use don't waste time for declare that. In NM I have to optimize performance and want to interact with `List Data Model` so I have to declare by self.

https://developer.android.com/reference/androidx/recyclerview/widget/ListAdapter